app/routes/admin/groups/components/GroupMembersList.tsx

```diff
@@ -47,21 +54,20 @@ const GroupMembersList = ({
   query,
   filters,
 }: Props) => {
+  // State for keeping track of which memberships are being edited
+  const [editState, setEditState] = useState(
+    memberships.reduce((acc, membership) => {
+      acc[membership.id] = false;
+      return acc;
+    }, {})
```

Comment:
Is this necessary, or could we just use an empty object where all ids would be undefined (still falsy)?

Reply:
But is it not the same? How would you write it?

Reply:
haha I started to doubt myself so I had to check in with the boss
<img width="651" alt="image" src="https://user-images.githubusercontent.com/69514187/230746102-27117100-4b21-4244-98c5-f6828f313699.png">

Reply:
When looking at how you use `editState` and `setEditState` I actually think this shouldn't be modeled this way at all 😬 . This logic creates an object with key-value pairs that, as you say: "_State for keeping track of which memberships are being edited_".  I feel like this is overly complex and it has a couple of bad effects. 

Firstly, you need this weird `reduce` to create your initial value and it needs to hold all memberships when you're probably only going to edit a couple.

Secondly, when you're using the `setEditState` you need to do the following 👇🏻 
```js
setEditState({
  ...editState,
  [id]: false,
});
// or
setEditState({
  ...editState,
  [membership.id]: !editState[membership.id],
})
```
which is more complex and error-prone than it should be. 

1. It requires you to always have the entire `editState` in scope, when in theory you're just toggling an id to `true` or `false`.
2. It requires you to know how to use `setEditState`. People might even update the value of a state using setState (_based on the previous state_) without taking the previous state via a function 😬 This is bad practice. You should always do `setEditState((prev) => {...prev, etc...})`

My suggestion is to just create a list of the `id's` gettings edited (_as this is basically what you're modeling here anyways_). Then you can use the `toggleEditId` method without any side effects and you don't need the state in scope to use it.

```ts
const [editState, setEditState] = useState([]) 
const toggleEditId = (id) => { 
  editState.includes(id)
    ? setEditState((current) => current.filter(filterId => filterId !== id))
    : setEditState((current) => ([...current, id]))
}

// Then later in the code whenever you check if we're editing an id
if (editState.includes(membership.id))

// And when you toggle something you just need the toggleEditId(id). 
onChange={async (value: { label: string; value: string }) => {
  toggleEditId(id)
  await editMembership(membership, value.value);
}}
```

Reply:
> Firstly, you need this weird reduce to create your initial value and it needs to hold all memberships when you're probably only going to edit a couple.

But the state does not hold all memberships? It only hold the memberships which are being edited. 

<img width="481" alt="image" src="https://user-images.githubusercontent.com/69514187/230771275-ec46e996-ef49-49d0-8f77-ac6402dc8de9.png">

I want to rename it from `editState` to `membershipsInEditMode` or something to make this clear.


---


> Secondly, when you're using the setEditState you need to do the following 👇🏻

That's literally how I do it? The previous state is taken in. I'm not sure if I'm following you here.

Reply:
If it starts as an empty list this is even more dumb! Why reduce nothing into nothing?

Reply:
> That's literally how I do it? The previous state is taken in. I'm not sure if I'm following you here.

No, you just spread `...editState` directly. You should take it as a function `(editState) => [...editState]`

Reply:
> No, you just spread ...editState directly. You should take it as a function (editState) => [...editState]

ah I misinterpreted your comment. I changed it.